xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -200,3 +450,128 @@ int CDVDSubtitlesLibass::GetNrOfEvents() const
     return 0;
   return m_track->n_events;
 }
+
+int CDVDSubtitlesLibass::AddEvent(std::string text, double startTime, double stopTime)
+{
+  if (text.empty())
+  {
+    CLog::Log(LOGDEBUG,
+              "{} - Add event skipped due to empty text (with start time: {}, stop time {})",
+              __FUNCTION__, startTime, stopTime);
+    return ASSEVENT_NO_ID;
+  }
+
+  CSingleLock lock(m_section);
+  if (!m_library || !m_track)
+  {
+    CLog::Log(LOGERROR, "{} - Missing ASS structs (m_library or m_track)", __FUNCTION__);
+    return ASSEVENT_NO_ID;
+  }
+
+  int eventId = ass_alloc_event(m_track);
+  if (eventId >= 0)
+  {
+    ASS_Event* event = m_track->events + eventId;
+    event->Start = DVD_TIME_TO_MSEC(startTime);
+    event->Duration = DVD_TIME_TO_MSEC(stopTime - startTime);
+    event->Style = m_track->default_style;
+    event->ReadOrder = eventId;
+    // Many subtitle formats use \r\n as line break,
```

Comment:
At this point you shouldn't touch the string anymore, that's a job for the parser or tagger, the string should be const

Reply:
done